### PR TITLE
flamenco; vm: VM Belt Sanding

### LINKED ISSUE #784: bn254
Upcoming Solana feature flags introduce the bn254 (aka alt_bn128) curve: "Barreto-Naehrig (BN) curve with 254-bit prime".

We need to support (de)compression, ADD, MUL, and PAIRING operations.

### PR DESCRIPTION
**Pull request by topointon-jump**
Kevin's, Emanuele's, Richie's and my fixes to the vm.

Ready for review now - will merge when all ledger tests are passing

Remaining TODOs and FIXMEs will be addressed in follow-up PRs.

Aggressive unit testing and fuzzing of the vm will ramp up after this is merged.

Closes #784

### CODE REVIEW

**Comment by ripatel-fd** [comment]
Converting this to draft due to merge conflicts.

**Review by ibhatt-jumptrading — APPROVED** [review]

File: src/app/fdctl/version.h [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#define FDCTL_MAJOR_VERSION 0UL
```
Comment (ravyu-jump) [inline src/app/fdctl/version.h:1]: this file is autogenerated and should not be checked in

**Review by ravyu-jump — CHANGES_REQUESTED** [review]

File: config/extra/with-tracing.mk [hunk 2]
```diff
@@ -1,3 +1,3 @@
 FD_DEBUG_SBPF_TRACES:=1
 CPPFLAGS+=-DFD_DEBUG_SBPF_TRACES=1
-CFLAGS+=-DFD_DEBUG_SBPF_TRACES=1
+CFLAGS+=-DFD_DEBUG_SBPF_TRACES=1
```
Comment (mmcgee-jump) [inline config/extra/with-tracing.mk:3]: Newline eof

File: contrib/test/run_test_vectors.sh [hunk 3]
```diff
@@ -58,4 +58,4 @@ then
 else
   echo 'test vector execution passed'
   exit 0
-fi
+fi
```
Comment (mmcgee-jump) [inline contrib/test/run_test_vectors.sh:61]: Newline eof

**Review by ravyu-jump — APPROVED** [review]

**Review by jumpsiegel — APPROVED** [review]

**Pull request merged**